src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -243,6 +243,28 @@ public function setRedrawFrequency(int $freq)
         $this->redrawFreq = max($freq, 1);
     }
 
+    /**
+     * Returns an iterator that will automatically update the progress bar when iterated.
+     *
+     * If $iterator is countable, the progress bar max steps will be set to its count.
+     */
+    public function iterate(iterable $iterable): iterable
+    {
+        if (\is_array($iterable) || $iterable instanceof \Countable) {
+            $this->setMaxSteps(\count($iterable));
+        }
+
+        $this->start();
+
+        foreach ($iterable as $key => $value) {
+            yield $key => $value;
+
+            $this->advance();
```

Comment:
shouldn't it be before the yield?

Reply:
If we put it before the yield, the progress bar will advance to 1 at the start of the loop wich means before the first element is processed which doesn't reflect the real progression.